lib/Sema/CSSimplify.cpp

```diff
@@ -5645,8 +5645,7 @@ bool ConstraintSystem::repairFailures(
 
     auto *parentLoc = getConstraintLocator(anchor, path);
 
-    if ((lhs->is<InOutType>() && !rhs->is<InOutType>()) ||
-        (!lhs->is<InOutType>() && rhs->is<InOutType>())) {
+    if (bool(lhs->is<InOutType>()) != bool(rhs->is<InOutType>())) {
```

Comment:
`TypeBase::is` already returns a `bool`, so this can just be `lhs->is<InOutType>() != rhs->is<InOutType>()`